app/helpers/application_helper.rb

```diff
@@ -1,27 +1,31 @@
 module ApplicationHelper
-  # rubocop:disable Style/OpenStructUse, Performance/OpenStruct
   USER_COLORS = ["#19063A", "#dce9f3"].freeze
 
-  DELETED_USER = OpenStruct.new(
-    id: nil,
-    darker_color: Color::CompareHex.new(USER_COLORS).brightness,
-    username: "[deleted user]",
-    name: "[Deleted User]",
-    summary: nil,
-    twitter_username: nil,
-    github_username: nil,
-  )
-  # rubocop:enable Style/OpenStructUse, Performance/OpenStruct
-
   LARGE_USERBASE_THRESHOLD = 1000
 
-  SUBTITLES = {
```

Comment:
Same as above, we can still keep a constant hash around, but the values will now come via I18n.

Reply:
Like this? (I didn't really test it out though)
``` ruby
  SUBTITLES = {
    "week" => -> { I18n.t("helpers.application_helper.subtitle.week") },
    "month" => -> { I18n.t("helpers.application_helper.subtitle.month") },
    "year" => -> { I18n.t("helpers.application_helper.subtitle.year") },
    "infinity" => -> { I18n.t("helpers.application_helper.subtitle.infinity") },
    "latest" => -> { I18n.t("helpers.application_helper.subtitle.latest") }
  }.freeze
```

Reply:
Ruby class bodies get evaluated on application start, so the values can be normal method calls:

```ruby
SUBTITLES = {
    "week" => I18n.t("helpers.application_helper.subtitle.week"),
    ...
```

Reply:
Perhaps I did it wrong in the past, but I'm afraid, if it only evaluated once on the app's start, the value will never change until the next startup. That means once it started (i.e. on the default locale), it sticks with it no matter viewers' locales are.

Reply:
As far as I understand, Rails doesn't fork a new app instance with each language, so the locale when this code is evaluated for the first time should be used forever.

Reply:
Yes, you're right, I was sure I was missing something here.